src/components/PostsList.tsx

```diff
@@ -1,86 +1,104 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { PostDetails } from './PostDetails';
+import classNames from 'classnames';
+import { getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  posts: Post[];
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({ posts }) => {
+  const [open, setOpen] = useState(true);
```

Comment:
The `open` state logic appears inverted. It should likely represent whether a post is open rather than closed. Consider renaming it to `isOpen` and adjusting the logic accordingly.